static/js/util.js

```diff
@@ -354,3 +354,126 @@ export function filter_by_word_prefix_match(items, search_term, item_to_text) {
 
     return filtered_items;
 }
+
+// ---------------- Get caret position inside a textbox ------------
+// Copied from http://jsfiddle.net/dandv/aFPA7
+// We use div to replicate the textarea exactly,
+// then get the position of text in that div.
+// We cannot get the position of caret in textarea
+// directly because it is a replaced element
+// (https://developer.mozilla.org/en-US/docs/Web/CSS/Replaced_element).
```

Comment:
I'm not sure we can use this code, since I don't see clear documentation of an open source license for it.  I'm not sure if there's a way to ask the author, but we might want to look for an OSS library that implements this function anyway.

Reply:
Replaced with https://github.com/deshiknaves/caret-pos which seems like a decent library.